### PR TITLE
[Substrate.io Migration] - Polkadot Protocol > Protocol Components > Transactions > Transaction and Block Basics

### PR DESCRIPTION
**Pull request by nhussein11**
This PR aims to migrate this page of the substrate tutorials: https://docs.substrate.io/learn/transaction-types/

### CODE REVIEW

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+# Transactions and Block Basis
+
+## Introduction
+
+In this article, you'll learn about the different types of transactions you can create and how to use them in a runtime. Transactions determine the data that makes its way into the blocks in your blockchain. By learning how different transaction types are used, you'll be better prepared to select the appropriate type for your needs.
+
+## What is a Transaction?
+
+In general, transactions provide a mechanism for making changes to a state that can be included in a block. There are three distinct transaction types in Substrate:
```
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:14]: Should we use Polkadot SDK or Substrate here? TBH, I am ok with using Substrate, as by now it should be obvious

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
+title: Transactions and Block Basis
+description: Overview of transactions and blocks in Polkadot SDK-Based Chains, including signed, unsigned, and inherent transactions.
+---
+
+# Transactions and Block Basis
+
+## Introduction
```
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:8]: Either we should add a note about the relationship between the term Extrinsic and Transaction, or just use Extrinsic IMO

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+The [`pallet_im_online::Call::heartbeat`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/im-online/src/lib.rs#L392){target=\_blank} extrinsic uses unsigned transactions to enable validator nodes to send a signal to the network to indicate that they are online. This function can only be called by a node registered as a validator in the network. The function includes internal logic to verify that the node is a validator, allowing the node to call the function using an unsigned transaction to avoid paying fees.
+
+### Inherent Transactions
+
+Inherent transactions —sometimes referred to as inherent—are a particular type of unsigned transaction. With this type of transaction, block authoring nodes can add information directly to a block. Inherent transactions can only be inserted into a block by the block authoring node that calls them. Typically, this type of transaction is not gossiped to other nodes or stored in the transaction queue. The data inserted using an inherent transaction is assumed valid without requiring specific validation.
```
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:40]:
```suggestion
Inherent transactions, sometimes referred to as inherent, are a particular type of unsigned transaction. With this type of transaction, block authoring nodes can add information directly to a block. Inherent transactions can only be inserted into a block by the block authoring node that calls them. Typically, this type of transaction is not gossiped to other nodes or stored in the transaction queue. The data inserted using an inherent transaction is assumed valid without requiring specific validation.
```

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+For example, if a block authoring node inserts a timestamp into a block, there is no way to prove that a timestamp is accurate. Instead, validators might accept or reject the block based on whether the timestamp is within some acceptable range of their system clocks.
+
+As an example, the [`pallet_timestamp::Call::now`](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/timestamp/src/lib.rs){target=\_blank} extrinsic enables a block authoring node to insert a current timestamp in each block the node produces. Similarly, the [`paras_inherent::Call::enter`](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/runtime/parachains/src/paras_inherent/mod.rs#L247){target=\_blank} extrinsic enables a parachain collator node to send its relay chain the validation data the relay chain expects.
+
+## What is a Block?
+
+In Substrate, a block consists of a header and an array of transactions. The header contains the following properties:
```
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:48]: Substrate or Polkadot SDK

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+- Block height - the number of blocks that have been created in the chain
+- Parent hash - the hash of the previous block in the chain
+- Transaction root - a cryptographic digest of the transactions in the block
+- State root - a cryptographic digest of the state after executing the transactions
+- Digest - a list of additional information that can be included in the block
+
+All of the transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transaction Lifecycle](TODO:update-path){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
+
+- It prevents any alterations to the series of transactions after the header has been built and distributed.
```
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:58]: Remove periods at end of list

**Review by CrackTheCode016 — COMMENTED** [review]
The content looks fine besides the highlighted comments.

Why are there some unused code snippets? Are they to be included in future documention?

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md (hunk 1 above)
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:14]: Let's stick to Substrate for now then, I agree :)

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md (hunk 4 above)
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:48]: Let's go with Substrate for now

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md (hunk 5 above)
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:58]: Nice catch!! 🚀

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+- Transaction root - a cryptographic digest of the transactions in the block
+- State root - a cryptographic digest of the state after executing the transactions
+- Digest - a list of additional information that can be included in the block
+
+All of the transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transaction Lifecycle](TODO:update-path){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
+
+- It prevents any alterations to the series of transactions after the header has been built and distributed
+- It enables light clients to succinctly verify that any given transaction exists in a block given only knowledge of the header
+
+## Where to Go Next
```
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basis.md:61]: I wouldn't say I like where to go next sections, but in this case, it seems useful. Feel free to decide about this

**Review by 0xLucca — APPROVED** [review]
LGTM

**Comment by eshaben** [comment]
Made some formatting updates and ran this through grammarly. Also replaced a couple of links that were pointing to specific lines in the SDK code base with links to the Rust docs, so we don't have to constantly watch for changes to the code to make sure that the line we were referring to is still the same line. @nhussein11  or @0xLucca can you please check that the new links make sense?

File: polkadot-protocol/protocol-components/transactions/.pages [hunk 7]
```diff
@@ -1,3 +1,4 @@
 title: Transactions
 nav:
   - index.md
+  - 'Transactions and Block Basis': transactions-and-block-basics.md
```
Comment (nhussein11) [inline polkadot-protocol/protocol-components/transactions/.pages:4]:
```suggestion
  - 'Transactions and Block Basics': transactions-and-block-basics.md
```

**Comment by nhussein11** [comment]
> Made some formatting updates and ran this through grammarly. Also replaced a couple of links that were pointing to specific lines in the SDK code base with links to the Rust docs, so we don't have to constantly watch for changes to the code to make sure that the line we were referring to is still the same line. @nhussein11 or @0xLucca can you please check that the new links are correct?

Thanks! I've checked those new links and they lgtm! I think this is good to be merged 🚀

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+In the Polkadot SDK, a block consists of a header and an array of transactions. The header contains the following properties:
+
+- **Block height** - the number of blocks that have been created in the chain
+- **Parent hash** - the hash of the previous block in the chain
+- **Transaction root** - a cryptographic digest of the transactions in the block
+- **State root** - a cryptographic digest of the state after executing the transactions
+- **Digest** - a list of additional information that can be included in the block
+
+All transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transactions Lifecycle](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle/){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
```
Comment (nhussein11) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md:56]:
```suggestion
All transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transactions Lifecycle](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle.md){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
```

**Comment by nhussein11** [comment]
This need approval from @polkadot-developers/docs-formatting-team to be merged

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+In the Polkadot SDK, a block consists of a header and an array of transactions. The header contains the following properties:
+
+- **Block height** - the number of blocks that have been created in the chain
+- **Parent hash** - the hash of the previous block in the chain
+- **Transaction root** - a cryptographic digest of the transactions in the block
+- **State root** - a cryptographic digest of the state after executing the transactions
+- **Digest** - a list of additional information that can be included in the block
+
+All transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transactions Lifecycle](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle.md){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
```
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md:56]:
```suggestion
All transactions are bundled together as a series to be executed as defined in the runtime. You'll learn more about transaction ordering in the [Transactions Lifecycle](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle/){target=\_blank} documentation. The transaction root is a cryptographic digest of this series. This cryptographic digest serves two purposes:
```

You don't need `.md` at the end of the links

File: polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+- It prevents any alterations to the series of transactions after the header has been built and distributed
+- It enables light clients to succinctly verify that any given transaction exists in a block given only knowledge of the header
+
+## Further Resources
+
+Now that you are familiar with transaction types and the information that constitutes a block, you can explore the following topics to learn more:
+
+- [**Transactions lifecycle**](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle.md){target=\_blank} - learn how transactions are processed and included in blocks
```
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-and-block-basics.md:65]:
```suggestion
- [**Transactions lifecycle**](/polkadot-protocol/protocol-components/transactions/transactions-lifecycle/){target=\_blank} - learn how transactions are processed and included in blocks
```

**Review by eshaben — COMMENTED** [review]
Just a couple links need to be fixed

**Review by dawnkelly09 — APPROVED** [review]
LGTM!

**Review by 0xLucca — APPROVED** [review]
LGTM

**Pull request merged**